beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -207,14 +208,14 @@ func (s *Service) validateRangeRequest(r *pb.BeaconBlocksByRangeRequest) error {
 // filters all the provided blocks to ensure they are canonical
 // and are strictly linear.
 func (s *Service) filterBlocks(ctx context.Context, blks []*ethpb.SignedBeaconBlock, roots [][32]byte, prevRoot *[32]byte,
-	step, startSlot uint64) ([]*ethpb.SignedBeaconBlock, error) {
+	step uint64, startSlot types.Slot) ([]*ethpb.SignedBeaconBlock, error) {
 	if len(blks) != len(roots) {
 		return nil, errors.New("input blks and roots are diff lengths")
 	}
 
 	newBlks := make([]*ethpb.SignedBeaconBlock, 0, len(blks))
 	for i, b := range blks {
-		isRequestedSlotStep := (b.Block.Slot-startSlot)%step == 0
+		isRequestedSlotStep := b.Block.Slot.SubSlot(startSlot).Mod(step) == 0
```

Comment:
this is very unlikely to happen(some issue indexing blocks in the db), but in the event the block's slot is less than start slot, we should fail gracefully. So use the `SafeSub` here.

Reply:
Will do!